compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BFiniteType.java

```diff
@@ -115,4 +116,16 @@ public void addValue(BLangExpression value) {
             nullable = true;
         }
     }
+
+    public Set<BType> getUniqueValues() {
+        Set<BType> uniqueType = new HashSet<>();
+        for (BLangExpression expression : this.valueSpace) {
+            if (expression.type.tag != TypeTags.FINITE) {
+                uniqueType.add(expression.type);
+            } else {
+                uniqueType.addAll(((BFiniteType) expression.type).getUniqueValues());
```

Comment:
Can you please share an example where the member in the value space of a finite type is another finite type?

Reply:
Think I have confused this with the union type